frontend/src/Search.elm

```diff
@@ -871,34 +907,91 @@ viewNoResults categoryName =
         ]
 
 
+closeButton : Html a
+closeButton =
+    span [] []
+
+
+viewBucket :
+    String
+    -> List AggregationsBucketItem
+    -> (String -> a)
+    -> List String
+    -> List (Html a)
+    -> List (Html a)
+viewBucket title buckets searchMsgFor selectedBucket sets =
+    List.append
+        sets
+        (if List.isEmpty buckets then
+            []
+
+         else
+            [ li []
+                [ ul []
+                    (List.append
+                        [ li [ class "header" ] [ text title ] ]
+                        (List.map
+                            (\bucket ->
+                                li []
+                                    [ a
+                                        [ href "#"
+                                        , onClick <| searchMsgFor bucket.key
+                                        , classList
+                                            [ ( "selected"
+                                              , List.member bucket.key selectedBucket
+                                              )
+                                            ]
+                                        ]
+                                        [ span [] [ text bucket.key ]
+                                        , if List.member bucket.key selectedBucket then
+                                            closeButton
+
+                                          else
+                                            span [] [ span [ class "badge" ] [ text <| String.fromInt bucket.doc_count ] ]
+                                        ]
+                                    ]
+                            )
+                            buckets
+                        )
+                    )
+                ]
+            ]
+        )
 viewSearchInput :
     (Msg a b -> c)
     -> String
-    -> String
+    -> Maybe String
     -> Maybe String
     -> Html c
 viewSearchInput outMsg categoryName selectedChannel searchQuery =
     form
         [ onSubmit (outMsg QueryInputSubmit)
         , class "search-input"
         ]
-        [ div []
+        ([]
+          |> List.append
```

Comment:
I'd like to move away from the `|> List.append` idiom as it inverts the order of the lists you're concatenating (and in this case breaks the frontend because the search input ends up below the channel selector).

Can you use `++` instead?

Reply:
Is there a reason to wrap things in `div []`?